### ISSUE TITLE
[bug]: query parameters are escaped implicitly

### ISSUE THREAD

**Issue by liudonghua123** [hoppscotch/hoppscotch#2829]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When I send a `Get` request with the following parameters(in bulk edit).

```
token: token_part1 token_part2
attrs: password%3Dabc
```

The actual request sent is `url?token=token_part1+token_part2&attrs=password%253Dabc` which is wrong for my server.

If I send the same `Get` request using postman, the actual request sent is `url?token=token_part1%20token_part2&attrs=password%3Dabc` which is correct for my server.

I found the problem code is around the following code.

https://github.com/hoppscotch/hoppscotch/blob/dc80cc80e6e9c02c41a0754dfa7868f7f102964a/packages/hoppscotch-app/src/helpers/strategies/ExtensionStrategy.ts#L64-L70

```js
url=new URL("http://localhost")
url.searchParams.append('token','token_part1 token_part2')
url.searchParams.append('attrs','password%3Dabc')
console.info(url.toString()) // http://localhost/?token=token_part1+token_part2&attrs=password%253abc

url=new URL('http://localhost/?token=token_part1+token_part2&attrs=password%253abc')
console.info(url.searchParams.get('token')) // token_part1 token_part2
console.info(url.searchParams.get('attrs')) // password%3Dabc
console.info(decodeURI('http://localhost/?token=token_part1+token_part2&attrs=password%253abc')) // http://localhost/?token=token_part1+token_part2&attrs=password%3Dabc


```

See also https://developer.mozilla.org/en-US/docs/Web/API/URL/toString, https://javascript.info/url, https://url.spec.whatwg.org/#URL-stringification-behavior.

### Steps to reproduce

1. Go to `Parameters` tab, click `Bulk edit`.
2. Fill with the following contents
  ```
  token: token_part1 token_part2
  attrs: password%3Dabc
  ```
3. Click `Send` button to make the request.

### Environment

Release

### Version

Cloud

**Comment by AndrewBastin** [comment]
Since the current implementation is correct according to the spec, we are prioritizing this as a low priority issue.

PR contributions for implementing the toggle for considering both ways are welcome in the meantime :)

**Comment by Rathan-Naik** [comment]
Just getting started on contributions @AndrewBastin I can work on this.

**Comment by AndrewBastin** [comment]
Cool @Rathan-Naik, please do let me know if you need any help/guidance.

Assigning the task to you.

**Comment by Ahishekoza** [comment]
Hello @AndrewBastin,
I would like to contribute to this issue

**Comment by nayanprasad** [comment]
Hi @liudonghua123 @AndrewBastin @Rathan-Naik @Ahishekoza @abraham . I would like to contribute to this issue. Let  me do it :)

**Comment by JManan** [comment]
hey @AndrewBastin i would like to contribute to this issue. Please let me do it.

**Comment by AndrewBastin** [comment]
@JManan sure.

**Comment by Pheonix075** [comment]
hey  @AndrewBastin  I would like to contribute to this issue.give me a chance

**Comment by VinayakSingh2001** [comment]
Hey @tejakummarikuntla I would like to work on this, can you please assign this to me !!

**Comment by tejakummarikuntla** [comment]
Sorry @VinayakSingh2001 , I don't think I can help you here. Did you mean to tag @AndrewBastin ?

**Comment by yashnirmal** [comment]
Hey @tejakummarikuntla is the issue still open?

**Comment by liudonghua123** [comment]
No PR currently? 

I suggest to add a configuration named Encode(Encode the query params, three options provided, enable means always encode the params, disable means do not encode the params and auto means encode the params which contain some common special characters need to encode).

And only two group of settings available now, one for `Theme`, another for `Interceptor`. Maybe a new group named `Other` is a good idea for this new `Encode` setting.

**Comment by meetdhanani17** [comment]
you need to add a toggle like space as %20 or + or i change to direct convert space as %20?

**Comment by maneeshms** [comment]
@tejakummarikuntla If this issue is still open and no PR raised yet, i would like to give a try.

**Comment by liyasthomas** [comment]
Assigning to @maneeshms.

**Comment by TheDynamicPunk** [comment]
Hey @liyasthomas it's been a while since there was some activity on this thread. If you've still not received any PR's for this, maybe I can take a look with some assistance from you?

**Comment by Vaibhav-tech22** [comment]
Hey @liyasthomas, I see there are no activities on this bug. If no PRs have been raised in this matter. I would like to give it a try. Thanks.

**Comment by mohitsourav7** [comment]
Hi @AndrewBastin I see there are no activities on this bug .If  you still haven't received any PR yet. I would like to give it a try 

Regards

**Comment by satwikkini-01** [comment]
Hello Mr @AndrewBastin , even I would like to give a try to solve this issue.
This will be my first open source contribution :)

**Comment by AndrewBastin** [comment]
@satwikkini-01 @mohitsourav7 if you are interested in opening a PR, feel free to do so, you do not need to wait on an assignment : )

**Comment by Pranay-Pandey** [comment]
Hey, @AndrewBastin 
I have raised a PR, with the changes suggested by @liudonghua123 .
But with these changes it seems the encoding takes place twice. 
![image](https://github.com/user-attachments/assets/eb3f2f38-4525-419c-9605-2907ee0a3f3a)
Any suggestions?

**Comment by jamesgeorge007** [comment]
#4412 Released in [v2024.10.0](https://github.com/hoppscotch/hoppscotch/releases/tag/2024.10.0).
